src/inmanta/plugins.py

```diff
@@ -421,6 +426,11 @@ def __call__(self, *args: object, **kwargs: object) -> object:
         The function call itself
         """
         self.check_requirements()
+        if self.deprecated:
```

Comment:
Perhaps we could move this before `check_requirements` so the user doesn't sink time in fixing requirements only to discover they should use a different plugin (which might not have these requirements).